gbpservice/nfp/service_vendor_agents/haproxy/haproxy_config_driver.py

```diff
@@ -498,4 +499,47 @@ def setup_ha(self, body):
                                         active_ip, standby_ip,
                                         priority, mgmt_interface)
 
+    def is_configured(self, ip, port):
+        config_to_check = "@" + ip + ":" + port
+        command = ['sudo', 'cat', HAPROXY_RSYSLOG_CONF_FILE_ABS_PATH]
+        status, output = commands.getstatusoutput(' '.join(command))
 
+        if status >= 0 and output:
+            for line in output.split('\n'):
+                if config_to_check in line:
+                    return True
+        return False
+
+    def configure_rsyslog_as_client(self, config):
+        OP_FAILED = { 'status' : False }
+        OP_SUCCESS = { 'status' : True }
+        try:
+            ip = config['server_ip']
+            port = str(config['server_port'])
+            if self.is_configured(ip, port):
+                return OP_SUCCESS
+            config_command = ['sed',
```

Comment:
Can we check if multiple time execution of this will not lead to duplicate entries? If already done, ignore this comment.